src/h3/to_multipoly.pyx

```diff
@@ -0,0 +1,88 @@
+cimport h3lib
+from h3lib cimport H3int
+from .util cimport check_addr
+
+
+# need to define `coord2geo` and `mercator` here due to Cython bug:
+# https://github.com/cython/cython/issues/2745
+cdef (double, double) mercator(double lat, double lng):
+    """Helper to coerce lat/lng range"""
+    lat = lat - 180 if lat > 90  else lat
+    lng = lng - 360 if lng > 180 else lng
+
+    return lat, lng
+
+
+cdef (double, double) coord2geo(h3lib.GeoCoord c):
+    return mercator(
+        h3lib.radsToDegs(c.lat),
+        h3lib.radsToDegs(c.lng)
+    )
+
+# todo: it's driving me crazy that these three functions are all essentially the same linked list walker...
+# grumble: no way to do iterators in with cdef functions!
```

Comment:
Is there a convenient way to define an iterator for these types so you can write a list comprehension here instead? Or does that not have much benefit?

Reply:
That's what I was really hoping to do! :)

The functions would be so clean:

```python
cdef walk_polys(const h3lib.LinkedGeoPolygon* L):
    while L:
        yield walk_loops(L.data)
        L = L.next
```

Unfortunately, a `cdef` function is essentially a C function, and [so it can't be a generator function (with a `yield` statement)](https://github.com/cython/cython/issues/2597).

We could turn it into a normal Python function to use the `yield` statement, and [Python functions **can** take in C pointers, but only if Cython can figure out how to do the conversion automatically](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#python-functions-vs-c-functions). It seems these objects are too complicated. It might still be possible, I just couldn't figure out how to do it.

The other option would be to crate a Python/Cython object to wrap the `h3lib.LinkedGeoPolygon*`, but at that point, it seemed to be creating more complexity than it would be saving...

Reply:
I tried something with [Cython fused types](https://cython.readthedocs.io/en/latest/src/userguide/fusedtypes.html), but couldn't get it working.

Will revisit later.